src/main/java/com/usharesoft/jenkins/steps/InstallStep.java

```diff
@@ -8,33 +10,85 @@
 import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream;
 import org.apache.commons.io.FileUtils;
 import org.apache.commons.io.IOUtils;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
 
+import java.io.BufferedReader;
 import java.io.File;
 import java.io.FileInputStream;
 import java.io.FileOutputStream;
 import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.StringReader;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.NoRouteToHostException;
 import java.net.URL;
+import java.util.Base64;
+
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
 
 import hudson.AbortException;
+import hudson.FilePath;
 import hudson.util.ArgumentListBuilder;
 
-public class InstallStep extends UForgeStep {
+public class InstallStep extends HammrStep {
     private String version;
 
-    public InstallStep(UForgeLauncher launcher, String version) {
-        super(launcher);
-        this.version = version;
+    public InstallStep(UForgeLauncher launcher, String url, StandardUsernamePasswordCredentials credentials, FilePath templatePath) {
+        super(launcher, url, credentials, templatePath);
     }
 
     @Override
     public void perform() throws InterruptedException, IOException {
         printStep(Messages.Logs_steps_install());
+        version = getUForgeVersion();
         downloadVirtualenv();
         decompress(launcher.getWorkspace() + "/virtualenv.tar.gz", new File(launcher.getWorkspace().getRemote()));
         launcher.launchInstall(getInitVenvCmd(), true);
         launcher.launchInstall(getInstallHammrCmd(), true);
     }
 
+    String getUForgeVersion() throws IOException {
+        try {
+            URL requestUrl = new URL(url + "/status");
+            HttpURLConnection connection = (HttpURLConnection) requestUrl.openConnection();
+            connection.setRequestMethod("GET");
+            connection.setRequestProperty("Accept", "application/xml");
+
+            String credentials = getUsername() + ":" + getPassword();
+            String encodedAuthorization = Base64.getEncoder().encodeToString(credentials.getBytes("UTF-8"));
+            connection.setRequestProperty("Authorization", "Basic "+ encodedAuthorization);
+
+            connection.setConnectTimeout(5000);
+            connection.setReadTimeout(5000);
+
+            handleStatus(connection.getResponseCode());
+
+            BufferedReader in = new BufferedReader(new InputStreamReader(connection.getInputStream(), "UTF-8"));
+            String inputLine;
+            StringBuilder content = new StringBuilder();
+            while ((inputLine = in.readLine()) != null) {
+                content.append(inputLine);
+            }
+
+            in.close();
```

Comment:
IMHO, I will move lines from new bufferedReader to this line (close()) in getVersionFromResponse(InputStreamReader).